modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/ManagementToolbarPropsTransformer.js

```diff
@@ -0,0 +1,252 @@
+
+		if (actionInputElement) {
+			actionInputElement.setAttribute('value', action);
+		}
+
+		const commandInputElement = form.querySelector(
+			`#${portletNamespace}cmd`
+		);
+
+		if (commandInputElement) {
+			commandInputElement.setAttribute('value', action);
+		}
+
+		submitForm(form, url, false);
+	};
+
+	const checkin = () => {
+		Liferay.componentReady(
+			`${portletNamespace}DocumentLibraryCheckinModal`
+		).then((documentLibraryCheckinModal) => {
+			documentLibraryCheckinModal.open((versionIncrease, changeLog) => {
+				const form = document.getElementById(`${portletNamespace}fm2`);
+
+				if (!form) {
+					return;
+				}
+
+				const changeLogInput = form.querySelector(
+					`#${portletNamespace}changeLog`
+				);
+
+				if (changeLogInput) {
+					changeLogInput.setAttribute('value', changeLog);
+				}
+
+				const versionIncreaseInput = form.querySelector(
+					`#${portletNamespace}versionIncrease`
+				);
+
+				if (versionIncreaseInput) {
+					versionIncreaseInput.setAttribute('value', versionIncrease);
+				}
+
+				processAction('checkin', editEntryURL);
+			});
+		});
+	};
+
+	const deleteEntries = () => {
+		let action;
+
+		if (trashEnabled) {
+			action = 'move_to_trash';
+		}
+		else if (
+			confirm(
+				Liferay.Language.get(
+					'are-you-sure-you-want-to-delete-the-selected-entries'
+				)
+			)
+		) {
+			action = 'delete';
+		}
+
+		processAction(action, editEntryURL);
+	};
+
+	const editTags = () => {
+		const searchContainer = Liferay.SearchContainer.get(
+			otherProps.searchContainerId
+		);
+
+		Liferay.componentReady(`${portletNamespace}EditTagsComponent`).then(
+			(editTagsComponent) => {
+				const bulkSelection = searchContainer.select?.get(
+					'bulkSelection'
+				);
+
+				const selectedFileEntries = searchContainer.select
+					.getAllSelectedElements()
+					.get('value');
+
+				editTagsComponent.open(
+					selectedFileEntries,
+					bulkSelection,
+					folderConfiguration.defaultParentFolderId
+				);
+			}
+		);
+	};
+
+	const move = () => {
+		const searchContainer = Liferay.SearchContainer.get(
+			otherProps.searchContainerId
+		);
+
+		let selectedItems = 0;
+
+		if (searchContainer.select) {
+			selectedItems = searchContainer.select
+				.getAllSelectedElements()
+				.filter(':enabled')
+				.size();
+		}
+
+		const dialogTitle =
+			selectedItems === 1
+				? Liferay.Language.get('select-destination-folder-for-x-item')
+				: Liferay.Language.get('select-destination-folder-for-x-items');
+
+		openSelectionModal({
+			height: '480px',
+			id: `${portletNamespace}selectFolder`,
+			onSelect(selectedItem) {
+				const newFolderId = selectedItem.folderid;
+
+				const form = document.getElementById(`${portletNamespace}fm2`);
+
+				if (!form) {
+					return;
+				}
+
+				form.setAttribute('action', editEntryURL);
+				form.setAttribute('enctype', 'multipart/form-data');
+				form.setAttribute('method', 'post');
+
+				const cmdInput = form.querySelector(`#${portletNamespace}cmd`);
```

Comment:
Here's me again, perhaps `getElementById` instead of `querySelector`.

Reply:
I always appreciate your (or anyone's) feedback so no problem.

I have feedback on your feedback, have you tried changing it to `form.getElementById`?

Reply:
Feedback on my feedback, that's my ultimate weakness! 
No I didn't try it 🤔  I'll try it to see why my feedback wasn't correct, that's the best way to learn!

Reply:
Both ``document.getElementById(`id`) `` and ``form.querySelector(`#id`)`` are fine, the performance difference is negligible.

`form.getElementById` would not work.

I personally have a small preference for ``document.getElementById(`id`) ``, because it promotes uniqueness of `id`s on a page. You would immediately get a bug and need to ensure uniquness. With ``form.querySelector(`#id`)``, you could merrily have working duplicate ids, that could bite you later down the line.

But this is minor, I would not consider ``form.querySelector(`#id`)`` to be a breaking issue.

Reply:
> I personally have a small preference for document.getElementById(`id`) , because it promotes uniqueness of ids on a page. You would immediately get a bug and need to ensure uniqueness. With form.querySelector(`#id`), you could merrily have working duplicate ids, that could bite you later down the line.

This would probably benefit easier writing of functional tests as well. If we had unique `id`'s on interactive or interested elements then it would be easier to maintain tests due to UI changes imo. One challenge QA has is to create robust and unique path locators but sometimes more than one element has the same html pattern and only thing that differentiates is text. Best case scenario for QA would be to have an easy to understand `id` for elements we want to test.